src/peft/tuners/lora/config.py

```diff
@@ -117,6 +119,25 @@ class LoraConfig(PeftConfig):
             )
         },
     )
+    if is_megatron_available():
+        megatron_config: Optional[ModuleType] = field(
```

Comment:
Is `megatron_config` really a `ModuleType`, isn't it a `TransformerConfig` from megatron? AFAICT, this is a `dataclass`. When we save the `adapter_config.json`, the `dataclass` will be converted to a dict/json. So when we load it back in, it's still a dict. We need to parse the `TransformerConfig` in that case.

Reply:
yes